diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/DataClassesRule.kt

```diff
@@ -61,6 +63,9 @@ class DataClassesRule(private val configRule: List<RulesConfig>) : Rule("data-cl
 
     @Suppress("UnsafeCallOnNullableType", "FUNCTION_BOOLEAN_PREFIX")
     private fun ASTNode.canBeDataClass(): Boolean {
+        if (findChildByType(PRIMARY_CONSTRUCTOR)?.let { constructor -> (constructor.psi as KtPrimaryConstructor).valueParameters.none { it.hasValOrVar() } } == true) {
```

Comment:
We should also check if there is at least one property, either in the primary constructor or in the class body. We are not checking it currently, but class without properties can't be a data class.